### ISSUE TITLE
extra.symfony.require installs packages outside range

### ISSUE THREAD

**Issue by garak** [symfony/flex#524]
I looked for similar issues, but I thought all of them have been closed with recent versions of flex.
With latest flex version (v1.3.1) and using `3.4.*` as value of `extra.symfony.require`, after composer installing or updating I get symfony/security: v4.2.10 installed.
I'm not sure of the reason why security component is allowing other Symfony components in version 3 (from 4.3 they're not allowed anymore) and maybe the bug is there, but anyway I'm not able to keep the same behaviour of good old symfony/lts package

**Comment by stof** [comment]
@fabpot `symfony/security` is missing on https://flex.symfony.com/versions.json

**Comment by fabpot** [comment]
@stof What you mean? I can see it:

<img width="328" alt="image" src="https://user-images.githubusercontent.com/47313/60264566-cca6ea80-98e3-11e9-96cb-f20c34216628.png">

**Comment by stof** [comment]
Indeed, I missed it. so this looks weird.

Btw, shouldn't we have `5.0` in there for components too ?

**Comment by stof** [comment]
@garak was Flex installed in your composer setup (either globally or in your project) at the time you ran `composer update` ?
The changes done in 1.3 solved issues for usage of `create-project` with our skeletons by adding the requirements only after installing Flex. But nothing can change the fact that a composer plugin cannot affect the dependency resolution of a project if it is not installed.

**Comment by garak** [comment]
Indeed, after issuing the update I see the following:

```
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 118 installs, 0 updates, 0 removals
  - Installing symfony/flex (v1.3.1): Loading from cache
Symfony operations: 1 recipe (377263e55e02d379f53f34ca5e391c26)
  - Configuring symfony/flex (>=1.0): From github.com/symfony/recipes:master
Loading composer repositories with package information
Updating dependencies (including require-dev)
Restricting packages listed in "symfony/symfony" to "3.4.*"
```

and then the updating is performed.

Anyway, I even tried to use a repository as skeleton, with `extra.symfony.require` configured as I mentioned above (and without lock file), it's not working as expected as well. Flex is also installed before other packages, but then constraint for 3.4 is not respected

**Comment by nicolas-grekas** [comment]
I cannot reproduce the issue, please provide exact steps to get it.

**Comment by fabpot** [comment]
`symfony/security` won't be available in 5.0.

**Comment by stof** [comment]
@fabpot sure, but others will (and no 5.0 is listed at all)

@garak for the reproducing steps, please provide the initial state of composer.json and of composer.lock (if there is any), as well of the vendor folder (in case it does not match the lock file), as well as commands you are running.

**Comment by garak** [comment]
[This is the repo](https://github.com/Bee-Lab/symfony-beelab/tree/garak) I use as skeleton, as mentioned before. As you can see, there's no lock files.
So, issuing `composer create-project beelab/symfony-beelab:dev-garak yourDir` flex is installed before, but then I end up with a lot of components in version 4.

**Comment by nicolas-grekas** [comment]
OK, here is the fix: https://github.com/Bee-Lab/symfony-beelab/pull/1

**Comment by garak** [comment]
@nicolas-grekas I just tried, I still get components in version 4:
```
garak@box:~$ composer create-project beelab/symfony-beelab:dev-garak foobar
Installing beelab/symfony-beelab (dev-garak 9dfd74d73e98526a6ed19c624b617d27a0165046)
  - Installing beelab/symfony-beelab (dev-garak garak): Cloning garak from cache
Created project in ciccio
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 112 installs, 0 updates, 0 removals
  - Installing symfony/flex (v1.3.1): Loading from cache
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 112 installs, 0 updates, 0 removals
  - Installing symfony/flex (v1.3.1): Loading from cache
  - Installing ramsey/uuid (3.8.0): Loading from cache
  - Installing psr/log (1.1.0): Loading from cache
  - Installing symfony/debug (v4.3.2): Loading from cache
  - Installing symfony/console (v3.4.29): Loading from cache
  - Installing doctrine/instantiator (1.2.0): Loading from cache
  - Installing doctrine/event-manager (v1.0.0): Loading from cache
  - Installing doctrine/cache (v1.8.0): Loading from cache
  - Installing doctrine/dbal (v2.9.2): Loading from cache
  - Installing doctrine/lexer (1.0.2): Loading from cache
  - Installing doctrine/annotations (v1.6.1): Loading from cache
  - Installing doctrine/reflection (v1.0.0): Loading from cache
  - Installing doctrine/collections (v1.6.2): Loading from cache
  - Installing doctrine/persistence (1.1.1): Loading from cache
  - Installing doctrine/inflector (v1.3.0): Loading from cache
  - Installing doctrine/common (v2.10.0): Loading from cache
  - Installing doctrine/orm (v2.6.3): Loading from cache
  - Installing ramsey/uuid-doctrine (1.5.0): Loading from cache
  - Installing symfony/yaml (v3.4.29): Loading from cache
  - Installing symfony/mime (v4.3.2): Loading from cache
  - Installing symfony/http-foundation (v4.3.2): Loading from cache
  - Installing symfony/event-dispatcher-contracts (v1.1.5): Loading from cache
  - Installing psr/container (1.0.0): Loading from cache
  - Installing psr/cache (1.0.1): Loading from cache
  - Installing symfony/event-dispatcher (v4.3.2): Loading from cache
  - Installing symfony/contracts (v1.1.0): Loading from cache
  - Installing symfony/http-kernel (v3.4.29): Loading from cache
  - Installing symfony/service-contracts (v1.1.5): Loading from cache
  - Installing symfony/dependency-injection (v3.4.29): Loading from cache
  - Installing symfony/filesystem (v4.3.2): Loading from cache
  - Installing symfony/config (v4.3.2): Loading from cache
  - Installing beberlei/assert (v3.2.1): Loading from cache
  - Installing simple-bus/message-bus (v3.0.2): Loading from cache
  - Installing simple-bus/symfony-bridge (v5.3.0): Loading from cache
  - Installing symfony/asset (v3.4.29): Loading from cache
  - Installing symfony/inflector (v4.3.2): Loading from cache
  - Installing symfony/property-access (v4.3.2): Loading from cache
  - Installing symfony/options-resolver (v4.3.2): Loading from cache
  - Installing symfony/intl (v4.3.2): Loading from cache
  - Installing symfony/form (v3.4.29): Loading from cache
  - Installing monolog/monolog (1.24.0): Loading from cache
  - Installing symfony/monolog-bridge (v4.2.10): Loading from cache
  - Installing symfony/monolog-bundle (v3.4.0): Loading from cache
  - Installing symfony/security (v4.2.10): Loading from cache
  - Installing symfony/security-bundle (v3.4.29): Loading from cache
  - Installing symfony/polyfill-iconv (v1.11.0): Loading from cache
  - Installing egulias/email-validator (2.1.9): Loading from cache
  - Installing swiftmailer/swiftmailer (v6.2.1): Loading from cache
  - Installing symfony/swiftmailer-bundle (v3.2.8): Loading from cache
  - Installing symfony/templating (v3.4.29): Loading from cache
  - Installing twig/twig (v2.11.3): Loading from cache
  - Installing symfony/translation-contracts (v1.1.5): Loading from cache
  - Installing symfony/twig-bridge (v4.0.15): Loading from cache
  - Installing symfony/twig-bundle (v3.4.29): Loading from cache
  - Installing symfony/translation (v4.3.2): Loading from cache
  - Installing symfony/validator (v3.4.29): Loading from cache
  - Installing twig/extensions (v1.5.4): Loading from cache
  - Installing doctrine/data-fixtures (v1.3.1): Loading from cache
  - Installing beelab/test-bundle (v2.1.2): Loading from cache
  - Installing symfony/doctrine-bridge (v4.2.10): Loading from cache
  - Installing doctrine/doctrine-cache-bundle (1.3.5): Loading from cache
  - Installing jdorn/sql-formatter (v1.2.17): Loading from cache
  - Installing symfony/routing (v4.3.2): Loading from cache
  - Installing symfony/finder (v4.3.2): Loading from cache
  - Installing symfony/class-loader (v3.4.29): Loading from cache
  - Installing symfony/var-exporter (v4.3.2): Loading from cache
  - Installing symfony/cache-contracts (v1.1.5): Loading from cache
  - Installing symfony/cache (v4.3.2): Loading from cache
  - Installing symfony/framework-bundle (v3.4.29): Loading from cache
  - Installing doctrine/doctrine-bundle (1.11.2): Loading from cache
  - Installing dama/doctrine-test-bundle (v5.0.3): Loading from cache
  - Installing doctrine/doctrine-fixtures-bundle (3.2.2): Loading from cache
  - Installing symfony/stopwatch (v3.4.29): Loading from cache
  - Installing symfony/process (v4.3.2): Loading from cache
  - Installing php-cs-fixer/diff (v1.3.0): Loading from cache
  - Installing composer/xdebug-handler (1.3.3): Loading from cache
  - Installing composer/semver (1.5.0): Loading from cache
  - Installing friendsofphp/php-cs-fixer (v2.15.1): Loading from cache
  - Installing sebastian/version (2.0.1): Loading from cache
  - Installing sebastian/type (1.1.2): Loading from cache
  - Installing sebastian/resource-operations (2.0.1): Loading from cache
  - Installing sebastian/recursion-context (3.0.0): Loading from cache
  - Installing sebastian/object-reflector (1.1.1): Loading from cache
  - Installing sebastian/object-enumerator (3.0.3): Loading from cache
  - Installing sebastian/global-state (3.0.0): Loading from cache
  - Installing sebastian/exporter (3.1.0): Loading from cache
  - Installing sebastian/environment (4.2.2): Loading from cache
  - Installing sebastian/diff (3.0.2): Loading from cache
  - Installing sebastian/comparator (3.0.2): Loading from cache
  - Installing phpunit/php-timer (2.1.2): Loading from cache
  - Installing phpunit/php-text-template (1.2.1): Loading from cache
  - Installing phpunit/php-file-iterator (2.0.2): Loading from cache
  - Installing theseer/tokenizer (1.1.3): Loading from cache
  - Installing sebastian/code-unit-reverse-lookup (1.0.1): Loading from cache
  - Installing phpunit/php-token-stream (3.0.1): Loading from cache
  - Installing phpunit/php-code-coverage (7.0.5): Loading from cache
  - Installing webmozart/assert (1.4.0): Loading from cache
  - Installing phpdocumentor/reflection-common (1.0.1): Loading from cache
  - Installing phpdocumentor/type-resolver (0.4.0): Loading from cache
  - Installing phpdocumentor/reflection-docblock (4.3.1): Loading from cache
  - Installing phpspec/prophecy (1.8.1): Loading from cache
  - Installing phar-io/version (2.0.1): Loading from cache
  - Installing phar-io/manifest (1.0.3): Loading from cache
  - Installing myclabs/deep-copy (1.9.1): Loading from cache
  - Installing phpunit/phpunit (8.2.3): Loading from cache
  - Installing symfony/dom-crawler (v4.3.2): Loading from cache
  - Installing symfony/browser-kit (v3.4.29): Loading from cache
  - Installing symfony/css-selector (v3.4.29): Loading from cache
  - Installing symfony/var-dumper (v4.3.2): Loading from cache
  - Installing symfony/debug-bundle (v3.4.29): Loading from cache
  - Installing symfony/phpunit-bridge (v3.4.29): Loading from cache
  - Installing symfony/web-profiler-bundle (v3.4.29): Loading from cache
```

**Comment by nicolas-grekas** [comment]
Got it, fixed in 6b14e1fb2b798cd9d7176a0fbb295d7da5c8fec7
Thanks for the details.
